### PR TITLE
[Enhancement] improve CompoundPredicateOperator hashcode performance

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

improve CompoundPredicateOperator hashcode performance

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/CompoundPredicateOperator.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CompoundPredicateOperator.java [hunk 1]
```diff
@@ -90,7 +89,13 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(opType, type, Sets.newHashSet(this.getChildren()).hashCode());
+        int h = 0;
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CompoundPredicateOperator.java:92]: is it better a hash function?

**Comment by liuyehcf** [comment]
> 

How much is the performance improvement?

**Review by liuyehcf — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CompoundPredicateOperator.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CompoundPredicateOperator.java:92]: it's a initialized value, we should keep same value every time call this function, and I think there is no need to use hash function, it also may have performance loss.

**Comment by Youngwb** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**